gammapy/image/radial_profile.py

```diff
@@ -0,0 +1,90 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Image utility functions"""
+import numpy as np
+from astropy.coordinates import Angle
+from astropy.wcs.utils import skycoord_to_pixel
+from astropy.table import QTable
+from . import SkyImage
+
+__all__ = [
+    'radial_profile_label_image',
+    'radial_profile',
+]
+
+
+def radial_profile_label_image(image, center, theta_bin=None):
+    """
+   Calculate the labeled image to know which pixel is group together for the radial profile.
+
+   Parameters
+   ----------
+   image : `~gammapy.image.SkyImage`
+         SkyImage of the image we cant to compute the radial profile
+   theta_bin : `~astropy.coordinates.Angle`
+         Bining for the radial profile
+   center : `~astropy.coordinates.SkyCoord`
+         The `SkyCoord` of the pixel used as the center. The default is
+            None, which then uses the coordinates of the center of the image.
+
+    Returns
+    -------
+    index_map : `~numpy.array`
+        label image
+   """
+    image_bin = Angle(np.fabs(image.meta["CDELT1"]), image.meta["CUNIT1"])
+    # Calculate the indices from the image
+    pix = image.coordinates_pix()
+    x = pix.x
+    y = pix.y
+    center_pix = skycoord_to_pixel(center, image.wcs, origin=0)
+    r = np.hypot(x - center_pix[0], y - center_pix[1])
+    if not theta_bin:
```

Comment:
I would remove this check here.

Raising an exception means users can't try to do this.
There might be cases where one wants to.

You could emit a warning, but I don't think it's needed.